js/app.jsx

```diff
@@ -486,6 +496,36 @@ var Job = React.createClass({
       </ul>
     );
 
+    function slimTrace(str) {
+      if (!str) {
+        return null;
+      }
+      var delimRegex = new RegExp('[,' + String.fromCharCode(255) + ']'); // [,ÿ]
+      var linesRegex = /(\w+)\.scala:(\d+)/i;
+      var lines = str.split(delimRegex).map(val => {
+        var trimmed = val.trim();
+        var matches = trimmed.match(linesRegex);
+        return matches ? matches[0] : trimmed;
+      });
+      // -1 if v1 < v2
+      var comparator = (v1, v2) => {
+        m1 = v1.match(linesRegex);
+        m2 = v2.match(linesRegex);
+        if (m1 && m2) {
+          return (m1[1] == m2[1]) ? (m1[2] - m2[2]) : m1[1].localeCompare(m2[1]);
+        }
+        return m1 ? 1 : -1;        
+      };
+      return (
+        <ul className="list-unstyled">
+          {unique(lines).sort(comparator).map(l => <li>{l}</li>)}
+        </ul>
```

Comment:
Can we track which lines are inputs and which are outputs and assign them to steps? Does scalding already do that? Tossing out a bunch of line numbers is halfway useful but it's a lot more useful if the user doesn't have to bounce around the code to figure out what's what.